src/components/dialogs/create-contingency-list-dialog.js

```diff
@@ -145,13 +171,22 @@ export const CreateContingencyListDialog = ({ open, onClose }) => {
             return;
         }
 
+        let formContent;
+        if (contingencyListType === ContingencyListType.FORM) {
+            formContent = currentCriteriaBasedFilter.current;
+        } else if (contingencyListType === ContingencyListType.SCRIPT) {
+            formContent = { script: currentScript };
+        }
+
         createContingencyList(
             contingencyListType,
             contingencyListName,
             contingencyListDescription,
+            formContent,
             activeDirectory
         )
             .then(() => {
+                setUnsavedChanges(false);
```

Comment:
maybe call handleCloseDialog instead of onClose here and add `setUnsavedChanges(false); in it as a cleaning step ?`